lib/portlayer/exec/container.go

```diff
@@ -133,28 +166,6 @@ func GetContainer(ctx context.Context, id uid.UID) *Handle {
 	return nil
 }
 
-func (s State) String() string {
-	switch s {
-	case StateCreated:
-		return "Created"
-	case StateStarting:
-		return "Starting"
-	case StateRunning:
-		return "Running"
-	case StateRemoving:
-		return "Removing"
-	case StateRemoved:
-		return "Removed"
-	case StateStopping:
-		return "Stopping"
-	case StateStopped:
-		return "Stopped"
-	case StateUnknown:
-		return "Unknown"
-	}
-	return ""
-}
-
 // CurrentState returns current state.
 func (c *Container) CurrentState() State {
 	c.m.Lock()
```

Comment:
That still leaves you open to torn updates within the Container as a whole - Runtime and Config need updating in a consistent manner.

We could change how they're stored and access, but don't think it's needed if we separate Handle and Container.